### DISCUSSION TITLE
Force VP8 video codec in the incoming offer during WebRTC ingestion

### DISCUSSION THREAD

**Discussion by gotodeneslaszlo** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1983] [Q&A] [answered]
Hello guys!

I am trying to find out how I can force VP8 video codec during WebRTC ingestion. At the moment, if I want to ingest video I invoke the JoinStorageSession API to start the WebRTC ingestion workflow, but the incoming offer always chooses H264 video codec which makes more preprocessing on the client side.

I tried to set the media type of the created video stream for ingestion, but the offer still uses H264, instead of VP8. The documentation says that VP8 is also supported.

Is it possible to force VP8 video codec in the incoming offer during WebRTC ingestion?

**Comment by disa6302** [comment, marked as answer]
@gotodeneslaszlo ,

VP8 ingestion is not supported. For WebRTC ingestion, only H264 is supported. However, in non-ingestion mode, VP8 is supported.

**Reply by gotodeneslaszlo** [reply]
The documentation and README.md do not mention this limitation neither. It would be better to explicitly define the supported codecs for WebRTC ingestion.

**Reply by disa6302** [reply]
You are correct. We have an internal ticket item to track this documentation miss. Thank you for bringing it up.

**Reply by gotodeneslaszlo** [reply]
Hi there! @disa6302 
I am curious about this: does WebRTC ingestion still only support H.264?

**Reply by sirknightj** [reply]
Currently, the storage session only supports H.264 ingest. You can reach out to this email (https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-js/discussions/320#discussioncomment-12405046) with your use case to submit a feature request.
